debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/JsonSerialization.java

```diff
@@ -53,16 +50,29 @@ public static interface Transformer extends Function<Document, String> {
     private final Transformer transformer;
 
     public JsonSerialization() {
-        final Encoder<Document> encoder = MongoClientSettings.getDefaultCodecRegistry().get(Document.class);
-        transformer = (doc) -> doc.toJson(COMPACT_JSON_SETTINGS, encoder);
+        transformer = (doc) -> doc.toJson(COMPACT_JSON_SETTINGS);
     }
 
-    public String getDocumentIdOplog(Document document) {
+    public String getDocumentIdOplog(BsonDocument document) {
         if (document == null) {
             return null;
         }
         // The serialized value is in format {"_": xxx} so we need to remove the starting dummy field name and closing brace
-        final String keyValue = new BasicDBObject("_", document.get(ID_FIELD_NAME)).toJson(SIMPLE_JSON_SETTINGS);
+        BsonValue value = document.get(ID_FIELD_NAME);
+        Object obj;
+        if (value.isObjectId()) {
+            obj = value.asObjectId().getValue();
+        }
+        else if (value.isInt32()) {
+            obj = value.asInt32().getValue();
+        }
+        else if (value.isInt64()) {
+            obj = value.asInt64().getValue();
+        }
+        else {
+            obj = value.asString().getValue();
```

Comment:
I feel a bit uneasy about this one. Any non-String `Id`  not covered by the cases above will end up thorowing an exception. And according to MongoDB any type byt array is supported to be used as `id` field.

Reply:
Good call, i tried BsonValue - seems it works with `BasicDBObject` (also from bson)!